template.tpl

```diff
@@ -927,46 +1033,84 @@ const generateConfiguration = () => {
   // Use manual configuration if it exists – otherwise use what was passed in the variable or an empty object
   const initOptions = (data.initOptions === 'manual' ? makeTableMap(manualOptions, 'key', 'value') : data.initOptions) || {};
 
-  // Configure for EU Data Residency
+  // Configuration for EU Data Residency
   if (data.euData) {
     initOptions.serverZone = 'EU';
   }
   
-  initOptions.attribution = {};
+  // Configuration for user agent enrichment plugin
+  if (!!data.uaParser) {
+    initOptions.uaParser = {
+      osName: data.osName,
+      osVersion: data.osVersion,
+      deviceManufacturer: data.deviceManufacturer,
+      deviceModel: data.deviceModel,
+    };
+  }
   
-  // Add attribution options
-  if (!!data.attribution) {
-    
-    if (!!data.attributionExcludeReferrers) {
-      initOptions.attribution.excludeReferrers = getType(data.attributionExcludeReferrers) === 'array' ? data.attributionExcludeReferrers : stringToArrayAndTrim(data.attributionExcludeReferrers);
+  if (!!data.defaultEventTracking) {
+    initOptions.defaultTracking = {};
+    if (!!data.detAttribution) {
+      initOptions.defaultTracking.attribution = {};
+        
+      if (!!data.attributionExcludeReferrers) {
+        // TODO: How to deal with regular expression input
```

Comment:
👍 Found a [post](https://support.google.com/tagmanager/thread/49104943?hl=en&msgid=49445274) in 2020 saying that templates didn't support regex. Not sure in 2023

Reply:
we can accept a string and convert to regex if for example it starts with `/`